arangod/Aql/DocumentProducingHelper.h

```diff
@@ -177,23 +178,22 @@ struct DocumentProducingFunctionContext {
   InputAqlItemRow const& _inputRow;
   OutputAqlItemRow* _outputRow;
   RegisterId const _outputRegister;
-  bool const _produceResult;
-  std::vector<std::string> const& _projections;
   transaction::Methods* const _trxPtr;
+  std::vector<std::string> const& _projections;
   std::vector<size_t> const& _coveringIndexAttributePositions;
-  bool _allowCoveringIndexOptimization;
+  bool const _produceResult;
   bool const _useRawDocumentPointers;
-  size_t _numScanned;
-
-  /// @brief set of already returned documents. Used to make the result distinct
-  std::unordered_set<TRI_voc_rid_t> _alreadyReturned;
-
+  bool _allowCoveringIndexOptimization;
   /// @brief Flag if the current index pointer is the last of the list.
   ///        Used in uniqueness checks.
   bool _isLastIndex;
 
   /// @brief Flag if we need to check for uniqueness
   bool _checkUniqueness;
+  size_t _numScanned;
+
+  /// @brief set of already returned documents. Used to make the result distinct
+  std::unordered_set<TRI_voc_rid_t> _alreadyReturned;
```

Comment:
shouldn't this be 
std::unord... 
size_t ...
bool ... 
here?

Reply:
why?
the crucial thing is that the boolean values are placed adjacently, as they can be 1-byte aligned.
all other members have an alignment of 8 bytes.

Reply:
ok, in this case its 5x bool, so the slack-space will be used anyways, however, I'd prefer to have them at the end, so its a non-brainer to add more there.

Reply:
You can have 5x anywhere, and you can add more.
The effect will be exactly the same.